packages/pyright-internal/src/tests/samples/dataclass26.py

```diff
@@ -0,0 +1,54 @@
+# This sample tests assignment of dataclass fields that use
+# the coverter parameter described in PEP 712.
+
+from dataclasses import dataclass, field
+from typing import Generic, TypeVar
+
+
+def converter_simple(s: str) -> int:
+    return int(s)
+
+
+@dataclass
+class Foo:
+    # This should generate an error because "converter" is not an official property yet.
+    field0: int = field(converter=converter_simple)
+
+foo = Foo("1")
+reveal_type(foo.field0, expected_text="int")
+foo.field0 = "2"
+
+# This should generate an error because only strs can be assigned to field0.
+foo.field0 = 2
+
+
```

Comment:
Another interesting test is to exercise the `__get__` and `__set__` methods by accessing the field from the class rather than a class instance. For example, what happens when you exec `Foo.field0 = ""`?

Reply:
I added tests for this, but it's not clear to me if the behavior is as expected.

```python
# Asymmetric
reveal_type(Foo.asymmetric, expected_text="int")
Foo.asymmetric = "2"
reveal_type(Foo.asymmetric, expected_text="int")
Foo.asymmetric = 2
reveal_type(Foo.asymmetric, expected_text="Literal[2]")

# Symmetric
reveal_type(Foo.symmetric, expected_text="str | int")
Foo.symmetric = "1"
reveal_type(Foo.symmetric, expected_text="Literal['1']")
```